.github/workflows/publish.yml

```diff
@@ -0,0 +1,36 @@
+name: Publish to S3
+
+on:
+  # Triggers the workflow on push events but only for the branches below
+  push:
+    branches: [ "main", "development" ]
+    paths:
+      - src/**
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks-out the repository under $GITHUB_WORKSPACE, so the job can access it
+      - uses: actions/checkout@v3
+      
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+           aws-access-key-id    :  ${{ secrets.AWS_CREDS_ACCESS_KEY }}
+           aws-secret-access-key:  ${{ secrets.AWS_CREDS_SECRET_KEY }}
+           aws-region           :  ${{ secrets.AWS_CREDS_REGION_NAME }}
+
+      # Syncs the src location to the S3 bucket
+      - name: Sync to S3
+        run: |
+           aws s3 sync src/ s3://${{ secrets.AWS_CONTENT_S3_BUCKET }}
```

Comment:
Will the contents of the main and dev branches be uploaded to separate locations in the target S3 bucket?

Reply:
Yes, it's tied to an environment secret for each branch that it's pushed to (filtered by a specific branch name), so that we don't have to change the variables for main and dev.

See here for details: https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment

Reply:
Got it. Just wanted to check the env for each branch was pointing to different locations.